index.php

```diff
@@ -89,9 +89,23 @@ public function __construct(string $baseDir) {
 		if (!file_exists($configFileName)) {
 			throw new \Exception('Could not find config.php. Is this file in the "updater" subfolder of Nextcloud?');
 		}
+		$filePointer = @fopen($configFileName, 'r');
+		if ($filePointer === false) {
+			throw new \Exception('Could not open config.php.');
+		}
+		if (!flock($filePointer, LOCK_SH)) {
+			throw new \Exception(sprintf('Could not acquire a shared lock on the config file'));
```

Comment:
```suggestion
			throw new \Exception(sprintf('Could not acquire a shared lock on the config file %s', $configFileName));
```

nit: Generating the exceptions consistently would be preferred (i.e. all with sprintf or all without, but not both).